### PR TITLE
Document support of `autoexporter` package

### LINKED ISSUE #437: Document autoexport package registration support
We should document that additional exporters can be registered via `autoexport` module.

_Originally posted by @pellared in https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/426#discussion_r1376264152_

### PR DESCRIPTION
**Pull request by MrAlias**
Fix #437 

- Update `WithEnv` docs to link to autoexporter pkg docs
- Add usage to CLI and include link  to autoexporter pkg docs
- Update README with link to where to find configuration info

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -130,6 +132,13 @@ To instrument an application running in Kubernetes, follow these steps:
 
 3. Deploy the application and the instrumentation using the manifest.
 
+### Configuration
+
+See the documentation for
+[`InstrumentationOption`](https://pkg.go.dev/go.opentelemetry.io/auto#InstrumentationOption)
+creation functions for information about how to configure the OpenTelemetry Go
+Automatic Instrumentation:
```
Comment (pellared) [inline README.md:140]:
```suggestion
See the documentation for
[`InstrumentationOption`](https://pkg.go.dev/go.opentelemetry.io/auto#InstrumentationOption)
factory functions for information about how to configure the OpenTelemetry Go
Automatic Instrumentation.
```

File: cli/main.go [hunk 2]
```diff
@@ -31,6 +33,25 @@ import (
...
+const help = `
+OpenTelemetry auto-instrumentation for Go applications using eBPF
+
+Environment variable configuration:
+
+	- OTEL_GO_AUTO_TARGET_EXE: sets the target binary
+	- OTEL_SERVICE_NAME (or OTEL_RESOURCE_ATTRIBUTES): sets the service name
+	- OTEL_TRACES_EXPORTER: sets the trace exporter
+
+The OTEL_TRACES_EXPORTER environment variable value is resolve using the
```
Comment (pellared) [inline cli/main.go:45]:
```suggestion
The OTEL_TRACES_EXPORTER environment variable value is resolved using the
```

File: instrumentation.go [hunk 3]
```diff
@@ -316,6 +316,10 @@ var lookupEnv = os.LookupEnv
 // and [WithServiceName] if their respective environment variable is defined.
 // If more than one of these options are used, the last one provided to an
 // [Instrumentation] will be used.
+//
+// The OTEL_TRACES_EXPORTER environment variable value is resolve using the
+// [autoexport] package. See that package's documentation for information on
```
Comment (pellared) [inline instrumentation.go:321]:
```suggestion
// [go.opentelemetry.io/contrib/exporters/autoexport] package. See that package's documentation for information on
```

File: instrumentation.go [hunk 4]
```diff
@@ -316,6 +316,10 @@ var lookupEnv = os.LookupEnv
 // and [WithServiceName] if their respective environment variable is defined.
 // If more than one of these options are used, the last one provided to an
 // [Instrumentation] will be used.
+//
+// The OTEL_TRACES_EXPORTER environment variable value is resolve using the
```
Comment (pellared) [inline instrumentation.go:320]:
```suggestion
// The OTEL_TRACES_EXPORTER environment variable value is resolved using the
```

File: instrumentation.go (hunk 3 above)
Comment (MrAlias) [reply instrumentation.go:321]: This currently uses the Go doc linking. The proposed change will not.

Comment (pellared) [reply instrumentation.go:321]: > When referring to other packages, “pkg” can be either a full import path or the assumed package name of an existing import.

I was not aware of it 👍

**Review by pellared — APPROVED** [review]

**Pull request merged**